src/bufr/BufrParser/BufrCollectors/BufrAccumulator.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * (C) Copyright 2020 NOAA/NWS/NCEP/EMC
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#include "BufrAccumulator.h"
+
+
+namespace Ingester
+{
+    BufrAccumulator::BufrAccumulator(Eigen::Index numColumns, Eigen::Index blockSize) :
+        dataArray_(blockSize, numColumns),
+        numColumns_(numColumns),
+        numDataRows_(0),
+        blockSize_(blockSize)
+    {
+    }
+
+    void BufrAccumulator::addRow(double* newRow)
+    {
```

Comment:
You should check for accidental calls of addRow with a nullptr. Maybe try gsl::not_null<double>. Or, better yet, try a gsl::span<double>, which ensures that the length is correctly passed.

Reply:
Now takes std::vector

Reply:
You should also make it a reference, otherwise copy-by-value will occur.